OpenEdXMobile/src/main/java/org/edx/mobile/base/BaseAppActivity.java

```diff
@@ -78,9 +81,19 @@ public boolean showGoogleCastButton() {
 
     @Override
     public void onCastStateChanged(int newState) {
-        if (mediaRouteMenuItem != null) {
-            googleCastDelegate.showIntroductoryOverlay(this, mediaRouteMenuItem);
+        // App initiate `IllegalArgumentException` when showing the IntroductoryOverlay in some cases,
+        // check following issue tracker for more details (still open).
+        // Ref: https://issuetracker.google.com/issues/36191274
+        // TODO: Replace the try-catch block with more appropriate logic / by updating the cast library
+        try {
+            if (isInForeground()) {
+                if (mediaRouteMenuItem != null) {
+                    googleCastDelegate.showIntroductoryOverlay(this, mediaRouteMenuItem);
+                }
+                invalidateOptionsMenu();
+            }
+        } catch (Exception e) {
+            logger.error(e);
```

Comment:
This won't log the exception to Crashlytics as a non-fatal, you'll have to call the function as follows:
` logger.error(e, true);`